### DISCUSSION TITLE
Question Regarding the Logic Behind Tailwind's Default Spacing Values

### DISCUSSION THREAD

**Discussion by onmax** [tailwindlabs/tailwindcss#11439] [General]
Hello Tailwind team and community 👋,

I'm a huge fan of Tailwind CSS for its incredible power and efficiency!

However, I stumbled across something interesting about the default width utility values, which at first glance seem non-intuitive. For example, `w-96`, which translates to `24rem`, caught my attention.

I suspect there may be some design wisdom or practical reasons behind this. 🤔 I'm eager to get to the bottom of it, but I have this question: Wouldn't it be more intuitive for newcomers(and also people with experience like me) if the values were directly equivalent (e.g. `w-20` for `20rem`)?

The current default values seem very random and against common sense for developers.

Any insight would be greatly appreciated! 🙏 Thanks for creating such a fantastic tool with Tailwind CSS!

**Comment by adamwathan** [comment]
Hey! The spacing scale was originally designed as a 4px scale, where every `1` unit represented 4px. So `pt-1` is 4px, `pt-2` is 8px, and `pt-36` is 114px. We use rem units under the hood so as to respect the preferences of people who have configured a default default font size other than 16px in their browser settings, but we think about them as pixels in our heads.

Originally we expected `4px` was the smallest amount we'd ever care to nudge an element by, but now almost 6 years later you can see that didn't work out, because we have values like `0.5`, `1.5`, `2.5`, and `3.5`. If I could go back in time I'd make `pt-1` be `1px` for sure.

**Reply by onmax** [reply]
Thanks for the explanation. Given the evolution and the needs of developers, I think it might be beneficial to introduce some additional flexibility into the spacing utilities. One suggestion would be to add a configuration property, perhaps called spacingUnits. This would allow users to map class values more directly to pixel values, making it more intuitive to work with.

Looking to the future, it might be a progressive step to make this the default configuration in TailwindCSS 4, streamlining the experience for developers.

But maybe this is a big change that you don't want to do, which I completely understand. Thanks again!
